pkg/operator/minio/controller.go

```diff
@@ -0,0 +1,301 @@
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coreos/pkg/capnslog"
+	opkit "github.com/rook/operator-kit"
+	miniov1alpha1 "github.com/rook/rook/pkg/apis/minio.rook.io/v1alpha1"
+	"github.com/rook/rook/pkg/clusterd"
+	"k8s.io/api/apps/v1beta2"
+	"k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+)
+
+// TODO: A lot of these constants are specific to the KubeCon demo. Let's
+// revisit these and determine what should be specified in the resource spec.
+const (
+	customResourceName       = "objectstore"
+	customResourceNamePlural = "objectstores"
+	minioPVCName             = "minio-pvc"
+	minioLabel               = "minio"
+	minioServerSuffixFmt     = "%s.svc.cluster.local"
+	minioVolumeName          = "data"
+	minioCtrName             = "minio"
+	minioCtrImage            = "minio/minio:RELEASE.2018-04-19T22-54-58Z"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "minio-op-object")
+
+// ObjectStoreResource represents the object store custom resource
+var ObjectStoreResource = opkit.CustomResource{
+	Name:    customResourceName,
+	Plural:  customResourceNamePlural,
+	Group:   miniov1alpha1.CustomResourceGroup,
+	Version: miniov1alpha1.Version,
+	Scope:   apiextensionsv1beta1.NamespaceScoped,
+	Kind:    reflect.TypeOf(miniov1alpha1.ObjectStore{}).Name(),
+}
+
+// MinioController represents a controller object for object store custom resources
+type MinioController struct {
+	context   *clusterd.Context
+	rookImage string
+}
+
+// NewMinioController create controller for watching object store custom resources created
+func NewMinioController(context *clusterd.Context, rookImage string) *MinioController {
+	return &MinioController{
+		context:   context,
+		rookImage: rookImage,
+	}
+}
+
+// StartWatch watches for instances of ObjectStore custom resources and acts on them
+func (c *MinioController) StartWatch(namespace string, stopCh chan struct{}) error {
+	resourceHandlerFuncs := cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onAdd,
+		UpdateFunc: c.onUpdate,
+		DeleteFunc: c.onDelete,
+	}
+
+	logger.Infof("start watching object store resources in namespace %s", namespace)
+	watcher := opkit.NewWatcher(ObjectStoreResource, namespace, resourceHandlerFuncs, c.context.RookClientset.MinioV1alpha1().RESTClient())
+	go watcher.Watch(&miniov1alpha1.ObjectStore{}, stopCh)
+
+	return nil
+}
+
+func (c *MinioController) makeMinioHeadlessService(name, namespace string, spec miniov1alpha1.ObjectStoreSpec, ownerRef meta_v1.OwnerReference) (*v1.Service, error) {
+	coreV1Client := c.context.Clientset.CoreV1()
+
+	svc, err := coreV1Client.Services(namespace).Create(&v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:            name,
+			Namespace:       namespace,
+			Labels:          map[string]string{"app": minioLabel},
```

Comment:
we have a `k8sutil.AppAttr` const you can use instead of using `"app"` directly.  this also applies to a few other places in this PR.